tiledb/sm/filter/filter_pipeline.cc

```diff
@@ -269,6 +270,15 @@ Status FilterPipeline::filter_chunks_reverse(
   // Run each chunk through the entire pipeline.
   auto statuses = parallel_for(0, chunks.size(), [&](uint64_t i) {
     const auto& chunk_input = chunks[i];
+
+    // We may skip reverse-filtering individual chunks.
+    const bool skip = std::get<4>(chunk_input);
+    if (skip) {
+      // We will leave garbage in 'output' at this chunk's offset because
+      // the reader will ignore this chunk.
```

Comment:
The idea here is that the output buffer size is unchanged (still allocating to the size of all uncompressed chunks), but we'll leave garbage in the ranges allocated for the ignored chunks. I am assuming that the upper stack will not try to read these ranges. I am not certain if that is true.

Reply:
A motivation behind selective chunk decompression is better memory management. So I think we need to rework this and handle it better properly saving memory wherever possible.

Reply:
We need to place the reverse-filtered chunks in a vector of buffers, which we can locate in constant time during cell copying.